### PR TITLE
Fix #1210 soft deleted posts with showthread.php?pid=x#pidx 2

### LINKED ISSUE #1210: soft deleted posts with showthread.php?pid=x#pidx
Default behaviour: Goto the corresponding thread and jump to post with pid x

Behaviour for hard deleted posts: Display error message (The specified thread does not exist.). 

Behaviour for soft deleted posts and user with right to see soft deleted posts: see default behaviour

Behaviour for soft deleted posts and user without right to see soft deleted posts: the default behaviour is used, so the user is shown the thread the post was in, but jumping to the right post using the # url fails (post is not displayed). Behaviour for hard deleted posts would be right in my point of view, but showing the thread and displaying an inline error that the specified post was deleted would also be ok.

Original thread: [soft deleted posts with showthread.php?pid=x#pidx](http://community.mybb.com/thread-156131.html)

### PR DESCRIPTION
**Pull request by Sama34**
Fix #1210 soft deleted posts with showthread.php?pid=x#pidx

I'm unsure why `$visibleonly` and `$visibleonly2` weren't set in the core for normal users?

https://github.com/Sama34/mybb/blob/fix-1210/showthread.php#L744

### CODE REVIEW

File: showthread.php [hunk 1]
```diff
@@ -120,7 +120,9 @@
 }
 else
 {
-	$ismod = false;
+	$ismod = false;	$ismod = false;
```
Comment (Destroy666x) [inline showthread.php]: Duplicate.

Also, this has nothing to do with that issue either, but whatever..

**Comment by Sama34** [comment]
It has everything to do with the issue @Destroy666x ...

With the new code implemented in the PR:
https://github.com/mybb/mybb/pull/1865/files#diff-7e30b0b016de305a3f054a785bc62fb6R50

The lastpost action can now result on a existing post which the user can't view. You get the jGrowl error message when landing then. Which makes zero sense.

If you check the code for lastpost action:
https://github.com/Sama34/mybb/blob/fix-1210/showthread.php#L265

You will see it checks for `$visibleonly` and `$visibleonly2` but those aren't set unless you are a moderator.

I come across this while debugging an issue with one of my plugins. A coincidence.

I think this makes the point of we needing more SQA members reviewing modifications.

**Comment by Destroy666x** [comment]
If you set `pid` parameter you don't set action to lastpost... So yes, not connected. But doesn't matter, good catch anyways.

**Comment by Sama34** [comment]
> So yes, not connected.

It is not directly  connected to the original issue but the fix to such issue causes this new.

That is the connection...

**Comment by Destroy666x** [comment]
> but the fix to such issue causes this new.

It didn't cause that issue - the code was wrong regardless of that fix. It only showed you that another bug existed.

**Pull request merged**